constructable rechargers

Segonist opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What does it change? What other things could this impact? -->
Now you can make cell recharger and recharger like other machines.
- Added circuits for cell recharger and recharger, they can be made in autholathe
- You can construct them like other machines, with 2 capacitors, and 5 lv cables for cell recharger and 5 mv cables for recherger.
- They can be upgraded (I don't think I made good formula for charging speed, correct me if so)
- Wall recharger wouldn't have it's construction graph, because I can't made it :(

**Media**
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/74138556/fee3467e-b30e-4172-9194-fb8d18b9027b

**Changelog**

<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged.

Only put changes that are visible and important to the player on the changelog.

Don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers

Putting a name after the :cl: symbol will change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB
-->

:cl: Segonist
- add: Now you can print cell recharger and recharger circuits on autolathe and construct them like other machines.

EmoGarbage404 commented on Resources/Prototypes/Catalog/Research/technologies.yml:
```diff
@@ -428,6 +428,8 @@
   - IndustrialEngineering
   unlockedRecipes:
   - PowerDrill
+  - CellRechargerCircuitboard
+  - WeaponCapacitorRechargerCircuitboard
```
these don't need to be a technology. just put them on the autolathe.

EmoGarbage404 commented on Resources/Prototypes/Entities/Objects/Devices/Circuitboards/Machine/production.yml:
```diff
@@ -447,6 +447,38 @@
+    - type: MachineBoard
+      prototype: PowerCellRecharger
+      requirements:
+        Capacitor: 2
```
don't use machine parts unless you're implementing upgrades. Mind you, upgrades would be less than 100 lines of code, so do consider it.

Segonist commented:
How can I edit this About the PR message?

deltanedas commented:
it feels wierd to be in autolathe when every other board in the game is in protolathe or circuit imprinter, even without technology requirement.

Segonist commented:
> it feels wierd to be in autolathe when every other board in the game is in protolathe or circuit imprinter, even without technology requirement.

should I just put them in the circuit imprinter?

deltanedas commented:
i think some like apc electronics are in both so maybe put in protolathe and circuit imprinter

Segonist commented:
> i think some like apc electronics are in both so maybe put in protolathe and circuit imprinter

wait, after rnd rework all boards from the protolathe were moved to the autolathe

Segonist commented:
```
Error Message:
   Multiple failures or warnings in test:
  1) Prototype PowerCellRecharger modifies component on spawn: ContainerContainer. Modified fields: containers
  2) Prototype WeaponCapacitorRecharger modifies component on spawn: ContainerContainer. Modified fields: containers
```
what does it mean and how to fix it?

deltanedas commented:
if you are using basemachine its probably double creating containers idk

deltanedas commented on Content.Server/Power/Components/ChargerComponent.cs:
```diff
@@ -8,8 +10,32 @@ public sealed class ChargerComponent : Component
         public CellChargerStatus Status;
 
+        /// <summary>
+        /// The charge rate of the charger
```
pro useless comment
at least say `, in watts`

deltanedas commented on Resources/Locale/en-US/power/components/charger.ftl:
```diff
@@ -1 +1,2 @@
-charger-examine = Charges [color={$color}]{$chargeRate}W[/color] per second.
\ No newline at end of file
+charger-examine = Charges [color={$color}]{$chargeRate}W[/color] per second.
```
change W to J or remove per second we are big boys and know what watts are :trollface:

Segonist replied:
what... why? It charges watts per second, isn't it?

deltanedas replied:
watts is joules per second bruh

DawBla commented on Resources/Prototypes/Entities/Structures/Power/chargers.yml:
```diff
@@ -93,10 +98,76 @@
+    mode: SnapgridCenter
   components:
   - type: Sprite
+    netsync: false
```
Sprite component is client only so no need for netsync: false
https://discord.com/channels/310555209753690112/928056457117696020/1102795784673579149

EmoGarbage404 commented:
These boards are good to be the autolathe under the doctrine of:  tech fpr basic shit you find on-station is dumb

deltanedas commented:
you find apcs, noors, firelocks, etc too
maybe all them should be in autolathe too

deltanedas commented:
im 99% sure they arent i checked like 2 hours ago

Segonist commented:
@metalgearsloth this pr is ready for review
